nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -1544,6 +1544,11 @@ def build_transformer_config(self) -> TransformerConfig:
 
         attention_softmax_in_fp32 = False  # not currently used in NeMo unless apply_query_key_layer_scaling is True
         apply_query_key_layer_scaling = self.cfg.get('apply_query_key_layer_scaling', False)
+
+        if apply_query_key_layer_scaling and not model_parallel_config.fp16:
+            logging.warning("apply_query_key_layer_scaling is only enabled when using FP16, setting it to False")
```

Comment:
I don't think model_parallel_config.fp16 is the right check though. That arg is for fp16 + megatron_amp_O2.

Maybe we should just check trainer.precision?

Reply:
can I check against `self.torch_dtype`? https://github.com/NVIDIA/NeMo/blob/8b27f3a68879a78d35fd7d5bac72fb792cf18da8/nemo/collections/nlp/models/language_modeling/megatron_base_model.py#L112